gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/EntityInteractionsFetcher.java

```diff
@@ -121,6 +121,96 @@ private List<String> getEntityIdColumnsFromInteraction(
     return columnNames;
   }
 
+  private boolean hasInteractionFilters(org.hypertrace.gateway.service.v1.common.Filter filter) {
+    if (ExpressionReader.isSimpleAttributeSelection(filter.getLhs())) {
+      String attributeId =
+          ExpressionReader.getAttributeIdFromAttributeSelection(filter.getLhs()).orElseThrow();
+
+      return !StringUtils.equals(attributeId, FROM_ENTITY_TYPE_ATTRIBUTE_ID)
+          && !StringUtils.equals(attributeId, TO_ENTITY_TYPE_ATTRIBUTE_ID);
+    }
+
+    boolean hasFilters = false;
```

Comment:
nit - avoid the variable and if check. We're adding extra lines of code and things to hold in our mind for no reason - pretty sure this is equivalent to these 6 lines:
```java
return filter.getChildFilterList().stream().anyMatch(this::hasInteractionFilters);
```